src/main/java/org/broadinstitute/hellbender/utils/fermi/FermiLiteAssembler.java

```diff
@@ -0,0 +1,161 @@
+package org.broadinstitute.hellbender.utils.fermi;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+public final class FermiLiteAssembler {
+    private static volatile boolean nativeLibLoaded = false;
+
+    public FermiLiteAssembler() {
+        loadNativeLibrary();
+    }
+
+    public interface BasesAndQuals {
+        byte[] getBases();
+        byte[] getQuals();
+    }
+
+    public FermiLiteAssembly createAssembly( final Iterable<BasesAndQuals> basesAndQuals ) {
+        return createAssembly(basesAndQuals, bAndQ -> bAndQ);
+    }
+
+    public <T> FermiLiteAssembly createAssembly( final Iterable<T> reads, final Function<T,BasesAndQuals> func ) {
+        final ByteBuffer assemblyData = createAssemblyData(makeReadData(reads, func));
+        if ( assemblyData == null ) throw new IllegalStateException("Unable to create assembly.");
+        try {
+            return interpretAssemblyData(assemblyData);
+        } finally {
+            destroyAssemblyData(assemblyData);
+        }
+    }
+
+    private static void loadNativeLibrary() {
+        if ( !nativeLibLoaded ) {
+            synchronized(FermiLiteAssembler.class) {
+                if ( !nativeLibLoaded ) {
+                    final String libNameOverride = System.getProperty("LIBFML_PATH");
+                    if ( libNameOverride != null ) {
+                        System.load(libNameOverride);
+                    }
+                    else {
+                        final String osName = System.getProperty("os.name", "unknown").toUpperCase();
+                        final String osArch = System.getProperty("os.arch");
+                        final String libName;
+                        if ( !"x86_64".equals(osArch) && !"amd64".equals(osArch) ) libName = null;
+                        else if ( osName.startsWith("MAC") ) libName = "/libfml.Darwin.dylib";
+                        else if ( osName.startsWith("LINUX") ) libName = "/libfml.Linux.so";
+                        else libName = null;
+                        if ( libName == null ) {
+                            throw new IllegalStateException("We have a JNI binding for fermi-lite only for x86-64 Linux and Mac.");
```

Comment:
If you set the LIBFML_PATH property incorrectly you'll see the error message when System.load fails.  I've made the messages for the exceptions when we can't figure out which pre-built binary to use much more explicit, suggesting that they may need to override.